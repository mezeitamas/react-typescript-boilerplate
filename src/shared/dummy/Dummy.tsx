import React from 'react';
import type { FunctionComponent, ReactElement } from 'react';

type DummyProps = {
    readonly name: string;
    readonly upperCase?: boolean;
};

const Dummy: FunctionComponent<DummyProps> = ({ name, upperCase }): ReactElement => {
    return <h1>Hello, {upperCase === true ? name.toUpperCase() : name}!</h1>;
};

export { Dummy };
