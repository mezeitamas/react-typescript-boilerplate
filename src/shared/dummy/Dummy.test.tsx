import React from 'react';

import ShallowRenderer from 'react-shallow-renderer';

import { Dummy } from './Dummy';

describe('<Dummy />', () => {
    test('should render', () => {
        // Given

        // When
        const renderer = new ShallowRenderer();
        const view = renderer.render(
            <Dummy
                name="Dummy"
                upperCase={true}
            />
        );

        // Then
        expect(view).toMatchSnapshot();
    });
});
