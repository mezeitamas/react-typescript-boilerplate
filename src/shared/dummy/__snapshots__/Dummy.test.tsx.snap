// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dummy /> should render 1`] = `
<h1>
  Hello, 
  DUMMY
  !
</h1>
`;
